### PR TITLE
Add reduce keyword to CrossEntropyLoss

### PR DESCRIPTION
**Pull request by zou3519**
### Summary
As per #264. `reduce=True` is the normal behavior, while `reduce=False` returns a loss per batch element.

### Test Plan
```
import torch
import torch.nn as nn
from torch.autograd import Variable
x = Variable(torch.randn(5, 6), requires_grad=True)
t = Variable(torch.zeros(5).random_(0, 6).long())
loss_reduce = nn.CrossEntropyLoss(size_average=False, reduce=True)
loss_no_reduce = nn.CrossEntropyLoss(reduce=False)
out_reduce = loss_reduce(x,t)
out_no_reduce = loss_no_reduce(x,t)
# assert that out_reduce, out_no_reduce.sum() are equal
# and that out_no_reduce is not scalar
```

### CODE REVIEW

**Review by apaszke — APPROVED** [review]

**Comment by apaszke** [comment]
Actually, we don't have tests for this, do we? Can you please add some?

**Comment by apaszke** [comment]
Ah it's just using NLLLoss that you PRed earlier. My bad. Thanks!

**Comment by napsternxg** [comment]
Any plans to backport this fix into version 0.2.0 already available via conda?

**Comment by apaszke** [comment]
It will be part of 0.3 release, which is going to be public this week

**Comment by napsternxg** [comment]
OK thanks. Does that mean, I can build using the 0.3 branch ? Is that branch stable?

**Comment by apaszke** [comment]
Yes, it's stable and that's what we're using to build the binaries right now

**Comment by napsternxg** [comment]
Thanks
